### ISSUE TITLE
need some optimize on find pathing

### ISSUE THREAD

**Issue by Amrf000** [Retera/WarsmashModEngine#21]
![image](https://user-images.githubusercontent.com/18072013/201517978-5bc9b4c3-6b3a-4b35-b617-3e7640a05c20.png)

**Comment by Retera** [comment]
Yes, definitely. I had not gone to the extent of using a profiler, but based on my experience testing the code and thinking about how I wrote it, this is absolutely the case. The pathfinding is currently the cause of most visual framedrops.

Do you have a pathfinding algorithm in mind that would retain the same visual fidelity of "feeling right" with regards to larger and smaller units in larger and smaller pathways, but that would perform much better?

For example, here is a video from a year and a half ago, when I was testing CPathfindingProcessor with regards to how small units like the penguin can travel to small pathways that a hero like Jaina cannot reach:
https://www.youtube.com/watch?v=L2YLW0J6qLE

So, the pathfinding has a lot of tweaks to try to make it fit individual cases. But those tweaks do not change the fact that I started from the pseudocode of A* search on Wikipedia: https://en.wikipedia.org//wiki/A*_search_algorithm

Maybe you have in mind a better pathfinding approach, instead? Perhaps you could implement it and put it into a pull request?

**Comment by Amrf000** [comment]
> Yes, definitely. I had not gone to the extent of using a profiler, but based on my experience testing the code and thinking about how I wrote it, this is absolutely the case. The pathfinding is currently the cause of most visual framedrops.
> 
> Do you have a pathfinding algorithm in mind that would retain the same visual fidelity of "feeling right" with regards to larger and smaller units in larger and smaller pathways, but that would perform much better?
> 
> For example, here is a video from a year and a half ago, when I was testing CPathfindingProcessor with regards to how small units like the penguin can travel to small pathways that a hero like Jaina cannot reach: https://www.youtube.com/watch?v=L2YLW0J6qLE
> 
> So, the pathfinding has a lot of tweaks to try to make it fit individual cases. But those tweaks do not change the fact that I started from the pseudocode of A* search on Wikipedia: https://en.wikipedia.org//wiki/A*_search_algorithm
> 
> Maybe you have in mind a better pathfinding approach, instead? Perhaps you could implement it and put it into a pull request?

I  will do some try
